### DISCUSSION TITLE
mogrify signifcantly slower after os upgrade

### DISCUSSION THREAD

**Discussion by z-108** [ImageMagick/ImageMagick#5496] [Help]
Hi, I recently upgraded my macos from Catalina --> Monterey. 
Now when I run a mogrify -format line that I've been using for a few years, I get CPU% from IM ranging from 100-400% and takes significantly longer to complete—a few minutes from previously seconds. 
My understanding of terminal commands is limited. I'm really open to suggestions as I'm hitting my head against the wall with this.

**I'm using:**
macOS Montery v.12.5.1
MacbookPro 2020
1.4 GHz Quad-Core
8 GB 2133 MHz

IM installed with brew
Version: ImageMagick 7.1.0-46 Q16-HDRI x86_64 20370 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib
Compiler: gcc (4.2)

**My command line is:**
cd [input folder]
mogrify -path [output folder] -format webp -delay 4 -loop 0 *

**I've tried:**
-limit thread 1 
--disable-openmp
neither seemed to improved high CPU or completion time
I've also tried only running the format command with mogrify and challenges still persist

**Comment by urban-warrior** [comment]
Your version is [HDRI](https://imagemagick.org/script/high-dynamic-range.php), perhaps that is the issue.  HDRI-enabled ImageMagick is floating point which can affect performance.  Install a non-HDRI version of ImageMagick and see if that increases performance.  Use this build command-line:

```
./configure --disable-hdri
```

If you cannot build ImageMagick yourself, try installing the ImageMagick version 6 with Brew.  It defaults to a non-HDRI install.

**Reply by z-108** [reply]
Thank you for your response! After building IM with HDRI disabled, CPU%'s have been below 100 so far. Mogrify seems to complete my tasks faster (down to about a minute per). It doesn't feel as fast as before upgrading my os which was around a few seconds to complete a task. Are there any suggestions for decreasing the completion time?

**Comment by urban-warrior** [comment]
Most of the CPU consumption is likely from WebP compression.  To decide, substitute PNM for WEBP as your format.  The difference in time is directly attributable to the WEBP image format processing.

**Reply by z-108** [reply]
Substituting PNM for WEBP did yield the task completion times that I'm looking for—completed them in a few seconds. Sadly, PNM is not support in the application these files are destined for. Thank you again for responding.

**Comment by urban-warrior** [comment]
See https://imagemagick.org/script/webp.php for WebP encoding options.  Perhaps one of them will increase performance.
